lib/superchain.ts

```diff
@@ -0,0 +1,19 @@
+import cbuild = require('@aws-cdk/aws-codebuild');
+import cdk = require('@aws-cdk/cdk');
+import path = require('path');
+import { LinuxPlatform } from './shellable';
+
+/**
+ * Expose the superchain docker linux build image as a construct.
+ */
+export class Superchain extends cdk.Construct {
+  public readonly platform: LinuxPlatform;
+
+  constructor(scope: cdk.Construct, id: string) {
```

Comment:
I am wondering why this actually needs an `id`. It should probably be a stack-singleton. I don't see any reason in the world for having more than one instance of those in the stack, so maybe we can implement it like this:

```ts
class Superchain extends LinuxPlatform {
  constructor(scope: cdk.Construct) {
    const stack = cdk.Stack.find(scope);
    const id = '<uuid>';
    const asset = stack.findChild(id) || cbuild.LinuxBuildImage.fromAsset(stack, id, { directory... });
    super(asset);
  }
}
```

Then, it can just be used as a LinuxPlatform.

Reply:
Cool! I was wondering what the best approach to this would be - completely agree.